daemon/images/image_list.go

```diff
@@ -149,6 +165,35 @@ func (i *ImageService) Images(_ context.Context, opts types.ImageListOptions) ([
 				summary.RepoTags = append(summary.RepoTags, reference.FamiliarString(ref))
 			}
 		}
+
+		// Ex. when reference = [debian, debain:x]
+		// even len(RepoTags) == 0, we should still output the summary
+		// But when reference = [debian:x, debian:y]
+		// we should not output the summary with len(RepoTags) == 0
```

Comment:
I don't think the described behaviour is correct. Multiple reference filters are ORed together (see #38171) so  it would be unexpected behaviour for `docker image ls --digests --filter reference=debian:x --filter reference=alpine` to include the repo digests for the `debian:x` image but not `docker image ls --digests --filter reference=debian:x`. Each reference filter should be considered independently; adding a reference filter value should not affect the output for images it doesn't match.